buildSrc/subprojects/cleanup/src/main/kotlin/org/gradle/cleanup/CleanupPlugin.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.cleanup
+
+import BuildHost
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.*
+import org.gradle.process.KillLeakingJavaProcesses
+
+class CleanupPlugin : Plugin<Project> {
+
+    override fun apply(project: Project): Unit = project.run {
+        tasks.create<CleanUpCaches>("cleanUpCaches") {
+            dependsOn(":createBuildReceipt")
+        }
+        tasks.create<CleanUpDaemons>("cleanUpDaemons")
+
+        val killExistingProcessesStartedByGradle by tasks.creating(KillLeakingJavaProcesses::class)
+
+//        if (BuildHost.isCiServer) {
+        if (!BuildHost.isCiServer) {
```

Comment:
It looks like this is reversed from the original script

Reply:
Good catch @big-guy!
@hansd this was merged and it may have made the CI situation worse by not cleaning up forked processes
I opened https://github.com/gradle/gradle/pull/4620 to restore previous behavior

Reply:
It was not merged like this. See the commit below.

Reply:
Sorry, it actually was. I don't know why. We are no longer using the BuildHost class. We are now using BuildEnvironment. I will change that. Thanks.